hail/python/hail/experimental/pca.py

```diff
@@ -0,0 +1,72 @@
+import hail as hl
+from hail.typecheck import *
+from hail.expr.expressions import *
+
+
+@typecheck(call_expr=expr_call,
+           loadings_expr=expr_array(expr_numeric),
+           af_expr=expr_numeric)
+def pc_project(call_expr, loadings_expr, af_expr):
+    """Projects genotypes onto pre-computed PCs. Requires loadings and
+    allele-frequency from a reference dataset (see example). Note that
+    `loadings_expr` must have no missing data and reflect the rows
+    from the original PCA run for this method to be accurate.
+
+    Example
+    -------
+    >>> # Compute loadings and allele frequency for reference dataset
+    >>> _, _, loadings_ht = hl.hwe_normalized_pca(mt.GT, k=10, compute_loadings=True)   # doctest: +SKIP
+    >>> mt = mt.annotate_rows(af=hl.agg.mean(mt.GT.n_alt_alleles()) / 2)                # doctest: +SKIP
+    >>> loadings_ht = loadings_ht.annotate(af=mt.rows()[loadings_ht.key].af)            # doctest: +SKIP
+    >>> # Project new genotypes onto loadings
+    >>> ht = pc_project(mt_to_project.GT, loadings_ht.loadings, loadings_ht.af)         # doctest: +SKIP
+
+    Parameters
+    ----------
+    call_expr : :class:`.CallExpression`
+        Entry-indexed call expression for genotypes
+        to project onto loadings.
+    loadings_expr : :class:`.ArrayNumericExpression`
+        Location of expression for loadings
+    af_expr : :class:`.Float64Expression`
+        Location of expression for allele frequency
+
+    Returns
+    -------
+    :class:`.Table`
+        Table with scores calculated from loadings in column `scores`
+    """
+    gt_source = call_expr._indices.source
+    loadings_source = loadings_expr._indices.source
+    af_source = af_expr._indices.source
+
+    check_entry_indexed('pc_project', gt_source)
```

Comment:
Tests are failing because you want these to be run directly on the expression, not the source. so 
```
    check_entry_indexed('pc_project', call_expr)
    check_row_indexed('pc_project', loadings_expr)
    check_row_indexed('pc_project', af_expr)
```